diff --git a/dist/cjs/index.cjs b/dist/cjs/index.cjs
index 5cc750bd99047779d6cb7b48559f567e8c3873cc..14098c564c993b0a5d1d4a745504b18725b6372c 100644
--- a/dist/cjs/index.cjs
+++ b/dist/cjs/index.cjs
@@ -92,8 +92,8 @@ function setupCoreRouterSsrQueryIntegration({ router, queryClient, dehydrateOpti
 			if (dehydrated.dehydratedQueryClient) (0, _tanstack_query_core.hydrate)(queryClient, dehydrated.dehydratedQueryClient, hydrateOptions);
 			const reader = dehydrated.queryStream.getReader();
 			reader.read().then(async function handle({ done, value }) {
-				(0, _tanstack_query_core.hydrate)(queryClient, value, hydrateOptions);
 				if (done) return;
+				(0, _tanstack_query_core.hydrate)(queryClient, value, hydrateOptions);
 				return handle(await reader.read());
 			}).catch((err) => {
 				console.error("Error reading query stream:", err);
diff --git a/dist/esm/index.js b/dist/esm/index.js
index bd4b864549e20e8c6b75b72ec096c55fc67376be..4546d9066757ebaabe7b7efbf79a52c07230e25f 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -91,8 +91,8 @@ function setupCoreRouterSsrQueryIntegration({ router, queryClient, dehydrateOpti
 			if (dehydrated.dehydratedQueryClient) hydrate(queryClient, dehydrated.dehydratedQueryClient, hydrateOptions);
 			const reader = dehydrated.queryStream.getReader();
 			reader.read().then(async function handle({ done, value }) {
-				hydrate(queryClient, value, hydrateOptions);
 				if (done) return;
+				hydrate(queryClient, value, hydrateOptions);
 				return handle(await reader.read());
 			}).catch((err) => {
 				console.error("Error reading query stream:", err);
diff --git a/src/index.ts b/src/index.ts
index 685653dea0c85c8a91e7a59aff25b8ed353c968e..0425728c3f7695a96ffc73be129f1a92f8fb169c 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -195,10 +195,10 @@ export function setupCoreRouterSsrQueryIntegration<TRouter extends AnyRouter>({
       reader
         .read()
         .then(async function handle({ done, value }) {
-          queryHydrate(queryClient, value, hydrateOptions)
           if (done) {
             return
           }
+          queryHydrate(queryClient, value, hydrateOptions)
           const result = await reader.read()
           return handle(result)
         })
